src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -23,7 +25,7 @@ public sealed class ReadOnlyCollectionBuilder<T> : IList<T>, System.Collections.
         private int _version;
 
         [NonSerialized]
-        private Object _syncRoot;
+        private object _syncRoot;
```

Comment:
https://msdn.microsoft.com/en-us/library/system.collections.icollection.syncroot(v=vs.110).aspx  says:

For collections whose underlying store is not publicly available, the expected implementation is to return the current instance. Note that the pointer to the current instance might not be sufficient for collections that wrap other collections; those should return the underlying collection's SyncRoot property.